src/app/components/AddDailyStatusComponent.jsx

```diff
@@ -69,23 +102,23 @@ class AddDailyStatusComponent extends React.Component {
           <div className="colWrapper">
             <div className="col2">
               <label>Date</label>
-              <select>{getOptions(this.state.dateOptions)}</select>
+              <select value={this.state.selDate} onChange={this.onChangeHandler.bind(this, "date")}>{getOptions(this.state.dateOptions)}</select>
```

Comment:
This binding is necessary to make `this` work in the callback. Now the binding also move to the constructor. All events are binded inside the constructor like:
this.onChangeHandler = this.onChangeHandler.bind(this);
this.onAddActivity = this.onAddActivity.bind(this);